### ISSUE TITLE
BE crash when turn on pipeline in ASAN mode

### ISSUE THREAD

**Issue by wangruin** [StarRocks/starrocks#4306]
### Steps to reproduce the behavior (Required)
1. CREATE TABLE `tpcds_100g_catalog_returns` (
  `cr_item_sk` int(11) NOT NULL COMMENT "",
  `cr_order_number` int(11) NOT NULL COMMENT "",
  `cr_returned_date_sk` int(11) NULL COMMENT "",
  `cr_returned_time_sk` int(11) NULL COMMENT "",
  `cr_refunded_customer_sk` int(11) NULL COMMENT "",
  `cr_refunded_cdemo_sk` int(11) NULL COMMENT "",
  `cr_refunded_hdemo_sk` int(11) NULL COMMENT "",
  `cr_refunded_addr_sk` int(11) NULL COMMENT "",
  `cr_returning_customer_sk` int(11) NULL COMMENT "",
  `cr_returning_cdemo_sk` int(11) NULL COMMENT "",
  `cr_returning_hdemo_sk` int(11) NULL COMMENT "",
  `cr_returning_addr_sk` int(11) NULL COMMENT "",
  `cr_call_center_sk` int(11) NULL COMMENT "",
  `cr_catalog_page_sk` int(11) NULL COMMENT "",
  `cr_ship_mode_sk` int(11) NULL COMMENT "",
  `cr_warehouse_sk` int(11) NULL COMMENT "",
  `cr_reason_sk` int(11) NULL COMMENT "",
  `cr_return_quantity` int(11) NULL COMMENT "",
  `cr_return_amount` decimal64(7, 2) NULL COMMENT "",
  `cr_return_tax` decimal64(7, 2) NULL COMMENT "",
  `cr_return_amt_inc_tax` decimal64(7, 2) NULL COMMENT "",
  `cr_fee` decimal64(7, 2) NULL COMMENT "",
  `cr_return_ship_cost` decimal64(7, 2) NULL COMMENT "",
  `cr_refunded_cash` decimal64(7, 2) NULL COMMENT "",
  `cr_reversed_charge` decimal64(7, 2) NULL COMMENT "",
  `cr_store_credit` decimal64(7, 2) NULL COMMENT "",
  `cr_net_loss` decimal64(7, 2) NULL COMMENT ""
) ENGINE=OLAP
DUPLICATE KEY(`cr_item_sk`, `cr_order_number`)
COMMENT "OLAP"
DISTRIBUTED BY HASH(`cr_item_sk`, `cr_order_number`) BUCKETS 5
PROPERTIES (
"replication_num" = "1",
"in_memory" = "false",
"storage_format" = "DEFAULT"
);
1. select   * from   tpcds_100g_catalog_returns as ref_0 where   ucase(     cast(       cast(         coalesce(           lpad(             cast(NULL as VARCHAR),             cast(ref_0.cr_call_center_sk as INT),             cast(NULL as VARCHAR)           ),           cast(null as VARCHAR)
     ) as VARCHAR       ) as VARCHAR     )   ) > right(     cast(       cast(         coalesce(           case             when cast(null as VARCHAR) >= cast(null as VARCHAR) then case               when true then space(                 cast(ref_0.cr_refunded_cdemo_sk as INT)               )             end             else case               when true then space(                 cast(ref_0.cr_refunded_cdemo_sk as INT)               )             end
   end,           cast(null as VARCHAR)         ) as VARCHAR       ) as VARCHAR     ),     cast(ref_0.cr_call_center_sk as INT)   );

### Expected behavior (Required)
Query returns correctly

### Real behavior (Required)
```
*** Check failure stack trace: ***
    @          0x9e67464  google::LogMessage::Fail()
    @          0x9e673c3  google::LogMessage::SendToLog()
    @          0x9e66d8c  google::LogMessage::Flush()
    @          0x9e6a058  google::LogMessageFatal::~LogMessageFatal()
    @          0x4304f58  starrocks::vectorized::BinaryColumn::~BinaryColumn()
    @          0x44e9112  __gnu_cxx::new_allocator<>::destroy<>()
    @          0x44e90e5  std::allocator_traits<>::destroy<>()
    @          0x44e8f03  std::_Sp_counted_ptr_inplace<>::_M_dispose()
    @          0x42e5cad  std::_Sp_counted_base<>::_M_release()
    @          0x42e2c1a  std::__shared_count<>::~__shared_count()
    @          0x44dca60  std::__shared_ptr<>::~__shared_ptr()
    @          0x44dca7c  std::shared_ptr<>::~shared_ptr()
    @          0x562408e  starrocks::vectorized::ColumnViewer<>::~ColumnViewer()
    @          0x88439c3  std::_Destroy<>()
    @          0x88405d2  std::_Destroy_aux<>::__destroy<>()
    @          0x8836962  std::_Destroy<>()
    @          0x8828259  std::_Destroy<>()
    @          0x880e4c4  std::vector<>::~vector()
    @          0x8ad5f9b  starrocks::vectorized::VectorizedCoalesceExpr<>::evaluate()
    @          0x4545c4c  starrocks::ExprContext::evaluate()
    @          0x8b67bbf  starrocks::vectorized::VectorizedFunctionCallExpr::evaluate()
    @          0x81fa5f7  starrocks::vectorized::VectorizedBinaryPredicate<>::evaluate()
    @          0x4545c4c  starrocks::ExprContext::evaluate()
    @          0x45459fd  starrocks::ExprContext::evaluate()
    @          0x6278116  starrocks::ExecNode::eval_conjuncts()
    @          0x74345f8  starrocks::pipeline::OlapChunkSource::_read_chunk_from_storage()
    @          0x7432a01  starrocks::pipeline::OlapChunkSource::buffer_next_batch_chunks_blocking()
    @          0x6f1c852  _ZZN9starrocks8pipeline12ScanOperator18_trigger_next_scanEPNS_12RuntimeStateEiENKUlvE0_clEv
    @          0x6f1fefc  _ZSt13__invoke_implIvRZN9starrocks8pipeline12ScanOperator18_trigger_next_scanEPNS0_12RuntimeStateEiEUlvE0_JEET_St14__invoke_otherOT0_DpOT1_
    @          0x6f1fd90  _ZSt10__invoke_rIvRZN9starrocks8pipeline12ScanOperator18_trigger_next_scanEPNS0_12RuntimeStateEiEUlvE0_JEENSt9enable_ifIX16is_invocable_r_vIT_T0_DpT1_EES8_E4typeEOS9_DpOSA_
    @          0x6f1f8d9  _ZNSt17_Function_handlerIFvvEZN9starrocks8pipeline12ScanOperator18_trigger_next_scanEPNS1_12RuntimeStateEiEUlvE0_E9_M_invokeERKSt9_Any_data
    @          0x430b32e  std::function<>::operator()()
```

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
branch-2.2 SQLSMITH_BRANCH_ASAN-b61b70b

**Comment by stdpain** [comment]
it was because function space generate a big binary column (greater than 4G). so offsets in binary column overflow.

**Comment by stdpain** [comment]
> F0322 11:25:07.999107 19753 binary_column.h:53] Check failed: _bytes.size() == _offsets.back() (4537990586 vs. 243023290)

**Comment by stdpain** [comment]
closed by #5159
